sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/subquery.scala

```diff
@@ -325,8 +325,6 @@ object PullupCorrelatedPredicates extends Rule[LogicalPlan] with PredicateHelper
    */
   def apply(plan: LogicalPlan): LogicalPlan = plan.transformUpWithPruning(
     _.containsPattern(PLAN_EXPRESSION)) {
-    case f @ Filter(_, a: Aggregate) =>
```

Comment:
This is not related to this PR but I just thought about the following case:
```sql
-- t1: [c1, c2]: (0,1),(1,2) , t: [a, b]: (0,1),(1,2)
select sum(c2) as c1 from t1 group by c1 having (select sum(b) from t where c1 = a) > 0
```
Because ResolveSubquery comes before ResolveAggregateFunction, this `c1` in `c1 = a` will be resolved as `sum(c2)` instead of `t1.c1`, because `UnresolvedHaving` is also a UnaryNode which will be resolved under the UnaryNode case below. So the result here can be incorrect.

Here we need to exclude `UnresolvedHaving` from this rule.